### ISSUE TITLE
Sky Nav behavior

### ISSUE THREAD

**Issue by tylersticka** [cloudfour/cloudfour.com-patterns#718]
## Overview

The `c-sky-nav` component exists but only includes markup and styles. This issue represents the remaining effort.

### Narrow viewports

- The `<nav>` should be hidden initially.
- Selecting the Menu button should toggle the visibility of the `<nav>`.
- When opening the nav, we should include an animation that makes it seem as if the browser is scrolling upward looking toward the sky (which is now populated by menu items), consistent with [our current navigation](https://cloudfour.com/).
- The inverse animation should be applied when closing the nav. This is also consistent with our current navigation.
- No animation should occur if the user prefers reduced motion.

### Wide viewports

- The `<nav>` should always be visible.

## Considerations

- To avoid odd scrolling behavior, our current solution applies a `translateY` transformation to the page contents to simulate that animation.
- Our current solution uses `will-change` on the `body` to avoid what was observed to be stuttering in Chrome on Android, but that seems like a poor solution that runs counter to most descriptions of `will-change`.

---

- [Components ▸ Sky Nav on Storybook](https://v-next--cloudfour-patterns.netlify.app/?path=/docs/components-sky-nav--dark)
- Previous Version: [Docs](https://cloudfour-patterns.netlify.app/patterns/components/sky.html), [Code](https://github.com/cloudfour/cloudfour.com-patterns/blob/master/src/assets/toolkit/scripts/lib/component/sky.js)
- HTML/CSS Issue: #479

**Comment by emersonthis** [comment]
> No animation should occur if the user prefers reduced motion.
Does this mean that the smooth scrolling to reveal the links doesn't happen; it just snaps open instantly? Is this "better" than the default behavior?

**Comment by spaceninja** [comment]
> Does this mean that the smooth scrolling to reveal the links doesn't happen; it just snaps open instantly?

Yes. When a user has the "prefers-reduced-motion" preference set, we should disable all animations. See: https://css-tricks.com/revisiting-prefers-reduced-motion-the-reduced-motion-media-query/

> Is this "better" than the default behavior?

Not sure what you mean by default behavior — the current behavior on `v-next` where the mobile nav is always open was not an intended state, but an incomplete feature awaiting a dev to add the toggle behavior.

**Comment by emersonthis** [comment]
What I'm trying to figure out is what "disable all animation" means in practice. Surely, we still want reduced motion users to be able to access the menu items. So if we're not "animating" them in/out of view, does that mean that we're instead "snapping" them instantly: I click "menu" and the menu _instantly_ appears? Or is there a different strategy, like making those items always visible, removing the need for a toggle?

More context: I might be asking a dumb question. I'm not super read-up on the literature about which kinds of users benefit from prefers-reduced-motion. My vague understanding is that people with certain types of cognitive conditions find animations/movement distracting/confusing. So it's not obvious to me whether an _instant_ reveal of the nav is actually better or worse for the likely users that would enable that preference.

**Comment by Paul-Hebert** [comment]
That's a good question!

I think we should snap it in instantly. There's some nuance to [whether all animations should be disabled or not](https://alistapart.com/article/designing-safer-web-animation-for-motion-sensitivity/) when `prefers-reduced-motion` is enabled, but in this case I think the safest thing to do is to completely remove the transition so the change happens instantly.

In the article linked above they mention that animations that move large amounts of content or move a long distance can be especially triggering for those with vestibular disorders. I think our sky nav animation definitely falls in that category.

[This article](https://css-tricks.com/introduction-reduced-motion-media-query/) mentions snapping to the final frame of a video instead of showing a full video which seems similar to what we'd be doing here. 

I think even users with vestibular disorders are generally okay with instantaneous content changes. (That's pretty much what happens whenever you open a new web page). I think it's the movement present in animation that can trigger the disorder.

I'd be super interested to learn more about this though! I've read a little about it but haven't done a deep dive so I'm curious what others think. 

@gerardo-rodriguez has this been covered yet in your a11y training?

I'd definitely be interested

**Comment by emersonthis** [comment]
Before putting this down, I encountered an issue with the storybook preview `<body>` starting with `transition: padding 0.3s ease 0s;`. This was thought to result from the use of [this storybook plugin](https://github.com/rbardini/storybook-addon-paddings) (with some more links [in Slack](https://cloudfour.slack.com/archives/C04AEEVK2/p1591820159123200?thread_ts=1591819880.122800&cid=C04AEEVK2)) but on closer inspection, it seems to already be disabled on the Sky Nav story, so its seems possible it's originating elsewhere.
